### PR TITLE
feat: ASAP-249 Migrate frontend to Vite

### PR DESCRIPTION
**Pull request by peterstarling**
This PR migrates CRA to Vite. The only remaining CRA bits are within the `contentful-app-extensions/` dir.

Notable changes:
* add gp2 to bundlewatch
* add gp2 release to Sentry
* fix some tests 
* migrate to Vite ;)

### CODE REVIEW

File: apps/crn-frontend/package.json [hunk 1]
```diff
@@ -6,11 +6,10 @@
     "node": "18.x"
   },
   "scripts": {
-    "start": "cross-env TSC_COMPILE_ON_ERROR=true react-scripts start",
-    "build": "node ./scripts/build-unsupported-browser-page.js && DISABLE_ESLINT_PLUGIN=true react-scripts build",
+    "start": "cross-env TSC_COMPILE_ON_ERROR=true vite start",
```
Comment (gabiayako) [inline apps/crn-frontend/package.json:9]: I believe we should have

```suggestion
    "start": "cross-env TSC_COMPILE_ON_ERROR=true vite",
```

according to this https://www.freecodecamp.org/news/how-to-migrate-from-create-react-app-to-vite/

Screenshot 2023-12-05 at 08 03 58

File: apps/gp2-frontend/package.json [hunk 2]
```diff
@@ -6,11 +6,10 @@
     "node": "18.x"
   },
   "scripts": {
-    "start": "cross-env TSC_COMPILE_ON_ERROR=true PORT=4000 react-scripts start",
-    "build": "node ./scripts/build-unsupported-browser-page.js && DISABLE_ESLINT_PLUGIN=true react-scripts build",
+    "start": "cross-env TSC_COMPILE_ON_ERROR=true PORT=4000 vite start",
```
Comment (gabiayako) [inline apps/gp2-frontend/package.json:9]: same as the comment above

```suggestion
    "start": "cross-env TSC_COMPILE_ON_ERROR=true PORT=4000 vite",
```

**Review by gabiayako — CHANGES_REQUESTED** [review]
I haven't looked at everything yet 😅 I'll come back later to look at the rest

**Review by gabiayako — APPROVED** [review]

File: apps/gp2-frontend/vite.config.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import react from '@vitejs/plugin-react';
+
+export default defineConfig({
+  plugins: [react()],
+  build: {
+    sourcemap: true,
+  },
+  server: {
+    open: true,
+    port: 3000,
```
[the remark this answers is not in the document]
Comment (peterstarling) [reply apps/gp2-frontend/vite.config.ts:11]: yes

File: apps/gp2-frontend/src/users/__tests__/UserDetail.test.tsx [hunk 4]
```diff
@@ -183,7 +183,8 @@ describe('UserDetail', () => {
       });
       mockGetUser.mockResolvedValueOnce(user);
       await renderUserDetail(user.id);
-      expect(screen.getByLabelText(/upload.+avatar/i)).toBeVisible();
+      expect(screen.getByLabelText(/upload.+avatar/i)).not.toBeVisible();
```
[the remark this answers is not in the document]
Comment (peterstarling) [reply apps/gp2-frontend/src/users/__tests__/UserDetail.test.tsx:186]: yeah we dont know why it was visible before in tests when it shouldn't, maybe @lctrt has an answer

**Pull request merged**